src/Sylius/Behat/Context/Ui/Admin/ManagingTaxRateContext.php

```diff
@@ -369,4 +382,48 @@ public function iShouldBeNotifiedThatTaxRateShouldNotEndBeforeItStarts(): void
     {
         $this->assertFieldValidationMessage('end_date', 'The tax rate should not end before it starts');
     }
+
+    /**
+     * @When /^I filter tax rates by (end|start) date from "(\d{4}-\d{2}-\d{2})"$/
+     */
+    public function iFilterTaxRatesByDateFrom(string $dateType, string $date): void
+    {
+        if ('start' === $dateType) {
+            $this->filterElement->specifyStartDateFrom($date);
+        } else {
+            $this->filterElement->specifyEndDateFrom($date);
+        }
```

Comment:
We can avoid this if...else by moving the logic of choosing the proper field by type to the element class